fuel-merkle/src/common/path_iterator.rs

```diff
@@ -139,7 +139,8 @@ where
         // 0      7     00  02  04  06  08  10  12  14   252 254
         //              00  01  02  03  04  05  06  07   126 127
         //
-        let initial_offset = T::key_size_in_bits() - root.height();
+        #[allow(clippy::arithmetic_side_effects)] // Responsibility of the trait implementor
+        let initial_offset = T::key_size_bits() - root.height();
```

Comment:
The size is not always the same. For instance `Position::key_size_bits() == 8` and `Node::key_size_bits() == 32`. Yes we could use `const KEY_SIZE_BITS: usize;` in the trait instead, if we know that the calculation can always be done in a const context.

Reply:
Got it. Then it sounds like if the implementer did something wrong, it could fail here? ANd maybe it is better to return an error then or at least a panic=)

Reply:
Yep, a panic would probably be in order.

Reply:
5ea05c09